src/OpenTelemetry.Collector.Dependencies/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -0,0 +1,127 @@
+{
+    internal class SqlClientDiagnosticListener : ListenerHandler
+    {
+        internal const string SqlDataBeforeExecuteCommand = "System.Data.SqlClient.WriteCommandBefore";
+        internal const string SqlMicrosoftBeforeExecuteCommand = "Microsoft.Data.SqlClient.WriteCommandBefore";
+
+        internal const string SqlDataAfterExecuteCommand = "System.Data.SqlClient.WriteCommandAfter";
+        internal const string SqlMicrosoftAfterExecuteCommand = "Microsoft.Data.SqlClient.WriteCommandAfter";
+
+        internal const string SqlDataWriteCommandError = "System.Data.SqlClient.WriteCommandError";
+        internal const string SqlMicrosoftWriteCommandError = "Microsoft.Data.SqlClient.WriteCommandError";
+
+        private readonly PropertyFetcher commandFetcher = new PropertyFetcher("Command");
+        private readonly PropertyFetcher connectionFetcher = new PropertyFetcher("Connection");
+        private readonly PropertyFetcher dataSourceFetcher = new PropertyFetcher("DataSource");
+        private readonly PropertyFetcher databaseFetcher = new PropertyFetcher("Database");
+        private readonly PropertyFetcher commandTextFetcher = new PropertyFetcher("CommandText");
+        private readonly PropertyFetcher exceptionFetcher = new PropertyFetcher("Exception");
+
+        public SqlClientDiagnosticListener(string sourceName, Tracer tracer)
+            : base(sourceName, tracer)
+        {
+        }
+
+        public override void OnStartActivity(Activity activity, object payload)
+        {
+        }
+
+        public override void OnCustom(string name, Activity activity, object payload)
+        {
+            switch (name)
+            {
+                case SqlDataBeforeExecuteCommand:
+                case SqlMicrosoftBeforeExecuteCommand:
+                    {
+                        var command = this.commandFetcher.Fetch(payload);
+
+                        if (command == null)
+                        {
+                            CollectorEventSource.Log.NullPayload($"{nameof(SqlClientDiagnosticListener)}-{name}");
+                            return;
+                        }
+
+                        var connection = this.connectionFetcher.Fetch(command);
+                        var database = this.databaseFetcher.Fetch(connection);
+
+                        this.Tracer.StartActiveSpan((string)database, SpanKind.Client, out var span);
+
+                        if (span.IsRecording)
+                        {
+                            var dataSource = this.dataSourceFetcher.Fetch(connection);
+                            var commandText = this.commandTextFetcher.Fetch(command);
+
+                            span.PutComponentAttribute("sql");
+
+                            span.PutDatabaseTypeAttribute("sql");
+                            span.PutDatabaseInstanceAttribute((string)database);
+                            span.PutDatabaseStatementAttribute((string)commandText);
```

Comment:
Seems like we read commandText always (for stored procedures as well as sql text), it'd be good to add comment here or todo here to sanitize the text (or make collection of sql text controlled by a setting in SqlClientCollectionOptions in the future.) as it can potentially contain secrets.

Reply:
@cijothomas Good call. I bit the bullet and added some basic options. Stored Procedure CommandText is captured by default. Text CommandText is NOT captured by default. Seemed like a much safer default behavior. Added `db.statementType` so you always get at least that. LMK your thoughts on all that.

Reply:
I agree this is right call - collect StoreProcedure name by default, collect SQL Command Text only if explicitly opted in via `SqlClientCollectionOptions`. 
Avoids privacy issues in the future.